### PR TITLE
go: Add error handling conventions

### PR DESCRIPTION
**Pull request by ohlsont**
Clarifications on how we use sad paths vs happy paths and general
error handling conventions.

### CODE REVIEW

File: go/README.md [hunk 1]
```diff
@@ -68,6 +68,59 @@ More context:
 > between unrelated concerns, encourage tight coupling and makes the code
 > that relies on them hard to test.
 
+### Error handling
+
+Generally error handling follow https://blog.golang.org/error-handling-and-go. We wrapp errors in order
```
Comment (odsod) [inline go/README.md]: * follows*
* wrap*
* adding*
* stack traces*
* (end paragraph with period)

File: go/README.md [hunk 2]
```diff
@@ -68,6 +68,59 @@ More context:
...
+to keep add context to the stacktraces
+
+More context:
+
+* [The xerror package](https://godoc.org/golang.org/x/xerrors)
```
Comment (odsod) [inline go/README.md]: xerrors*

File: go/README.md [hunk 3]
```diff
@@ -68,6 +68,59 @@ More context:
...
+
+#### Happy path vs Sad path
+
+In the normal case the code returns errors trough sad paths branchouts where errors are handled and returned.
```
Comment (odsod) [inline go/README.md]: * through*

File: go/README.md [hunk 4]
```diff
@@ -68,6 +68,59 @@ More context:
...
+Happy paths keeps to the left edge of the code
```
Comment (odsod) [inline go/README.md]: * end sentence with period

File: go/README.md [hunk 5]
```diff
@@ -68,6 +68,59 @@ More context:
...
+
+```go
+// good
+func foo() error {
+        err := bar()
+        if err != nil {
+                // sad path handles and returns errors
+                return xerror.Errorf("foo: %w", err)
+        }
+        // happy paths return non-errornous results
```
Comment (odsod) [inline go/README.md]: * erroneous*

File: go/README.md [hunk 6]
```diff
@@ -68,6 +68,59 @@ More context:
...
+func foo() error {
+        err := bar()
+        if err != nil {
+                return xerror.Errorf("foo: %w", err)
+        } else {
+                return nil
+        }
+}
+```
+
```
Comment (odsod) [inline go/README.md]: remove blank line

File: go/README.md [hunk 7]
```diff
@@ -68,6 +68,57 @@ More context:
...
+#### Happy path vs Sad path
+
+In the normal case the code returns errors through sad paths branch outs where errors are handled and returned.
+Happy paths keeps to the left edge of the code.
+
+```go
+// good
+func foo() error {
+        err := bar()
+        if err != nil {
```
Comment (odsod) [inline go/README.md]: make this an inline statement to show more best practices at the same time:

~~~go
if err := bar(); err != nil {
}
~~~

Comment (ohlsont) [reply go/README.md]: fixed :)

**Pull request merged**